core/adapters/eth_tx.go

```diff
@@ -42,6 +44,85 @@ func (e *EthTx) TaskType() models.TaskType {
 // is not currently pending. Then it confirms the transaction was confirmed on
 // the blockchain.
 func (e *EthTx) Perform(input models.RunInput, store *strpkg.Store) models.RunOutput {
+	if store.Config.EnableBulletproofTxManager() {
+		return e.perform(input, store)
+	} else {
+		return e.legacyPerform(input, store)
+	}
+}
+
+// TODO(sam): Move away from resuming this on every new head and do something more sensible
+func (e *EthTx) perform(input models.RunInput, store *strpkg.Store) models.RunOutput {
+	trtx, err := store.FindEthTaskRunTxByTaskRunID(input.TaskRunID().UUID())
+	if err != nil {
+		logger.Error(err)
```

Comment:
Can we add some context here `logger.Error(err)` tends to be very challenging to use.

Reply:
Yeah, I thought about it but figured the stacktrace would be enough.

Reply:
Perhaps you could give an example of what you have in mind?

Reply:
Generally I like to have something like `logger.Error("error finding task by task run id", err)`

Reply:
Yup, I can do this